### ISSUE TITLE
Support all HPA versions

### ISSUE THREAD

**Issue by mrunesson** [stackrox/kube-linter#287]
**Description of the problem/feature request**
Current implementation of template hpa-minimum-replicas only support v2beta1 of horizontalpodautoscaler. There is also a stable version v1 and a newer beta v2beta2. It would be valuable to support these two versions too.

**Description of the existing behavior vs. expected behavior**
This is tested on head of main branch.

The following is generating the expected warning of too few replicas.
```yaml
apiVersion: autoscaling/v2beta1
kind: HorizontalPodAutoscaler
metadata:
  name: demo
spec:
  maxReplicas: 5
  minReplicas: 2
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: demo-app
  targetCPUUtilizationPercentage: 50
```

Changing the version to v1 lead to a false statement of `No lint errors found!`:
```yaml
apiVersion: autoscaling/v1
kind: HorizontalPodAutoscaler
metadata:
  name: demo
spec:
  maxReplicas: 5
  minReplicas: 2
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: demo-app
  targetCPUUtilizationPercentage: 50
```

The expected output of this would be:
```text
hpa.yaml: (object: aardvark-demo-develop/aardvark-demo autoscaling/v2beta1, Kind=HorizontalPodAutoscaler) object has 2 replicas but minimum required replicas is 3 (check: hpa-minimum-three-replicas, remediation: Increase be number of replicas in the HorizontalPodAutoscaler to at least three to increase fault tolerance.)
```

Same for v2beta2.

**Comment by janisz** [comment]
Good catch. It looks like problem is in 

It should be handled with all possible versions. Something like this should solve the issue

https://github.com/stackrox/kube-linter/blob/693aae0bf6beb10d2dbf7bdcf72b28ce4aa99d31/pkg/extract/hpa_spec.go#L8-L18


```go
package extract

import (
	"golang.stackrox.io/kube-linter/pkg/k8sutil"
	autoscalingV1 "k8s.io/api/autoscaling/v1"
	autoscalingV2 "k8s.io/api/autoscaling/v2"
	autoscalingV2Beta1 "k8s.io/api/autoscaling/v2beta1"
	autoscalingV2Beta2 "k8s.io/api/autoscaling/v2beta2"
)

// HPAMinReplicas extracts minReplicas from the given object, if available.
func HPAMinReplicas(obj k8sutil.Object) (int32, bool) {
	switch hpa := obj.(type) {
	case *autoscalingV2Beta1.HorizontalPodAutoscaler:
		return checkReplicas(hpa.Spec.MinReplicas)
	case *autoscalingV2Beta2.HorizontalPodAutoscaler:
		return checkReplicas(hpa.Spec.MinReplicas)
	case *autoscalingV2.HorizontalPodAutoscaler:
		return checkReplicas(hpa.Spec.MinReplicas)
	case *autoscalingV1.HorizontalPodAutoscaler:
		return checkReplicas(hpa.Spec.MinReplicas)
	default:
		return 0, false
	}
}

func checkReplicas(minReplicas *int32) (int32, bool) {
	if minReplicas != nil {
		return *minReplicas, true
	}
	// If numReplicas is a `nil` pointer, then it defaults to 1.
	return 1, true
}
```

@mrunesson Would you like to contribute a patch for it?

**Comment by mrunesson** [comment]
I am happy to give it a try. I am pretty new to go but want to learn.

**Comment by janisz** [comment]
It was fixed by #290
